src/amo/components/SearchFilters/index.js

```diff
@@ -105,15 +105,24 @@ export class SearchFiltersBase extends React.Component<InternalProps> {
   doSearch(newFilters: SearchFiltersType) {
     const { clientApp, lang, history, pathname } = this.props;
 
-    if (newFilters.page) {
+    const filters = { ...newFilters };
+
+    if (filters.page) {
       // Since it's now a new search, reset the page.
       // eslint-disable-next-line
-      newFilters.page = '1';
+      filters.page = '1';
+    }
+
+    // If category is a filter, remove category and type as they are already
+    // included in pathname.
+    if (filters.category) {
```

Comment:
Good point. It shouldn't ever be needed.

Reply:
I think this might be the comment that made me remove the code that removes the category and type filters, but I think we did in fact need that code, as it was used to remove them when a user changes the sort sequence from the SearchFilters component. I'll add it back in and see if that fixes it.